### PR TITLE
Ports: Add Pacman port

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This adds a port for https://github.com/ebuc99/pacman:

Screenshot from 2022-10-23 12-58-33

### CODE REVIEW

File: Ports/pacman/package.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=pacman
+version=git
+workdir=pacman-master
```
Comment (timschumi) [inline Ports/pacman/package.sh:4]: These are strings without variable expansions, so they should be in single-quotes.

```suggestion
port='pacman'
version='git'
workdir='pacman-master'
```

File: Ports/pacman/package.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+files="https://github.com/ebuc99/pacman/archive/refs/heads/master.tar.gz pacman-master.tar.gz 3016af69e168e0dd87a079bbfb35d995cf2e7bc1b710716c81ac99323bb4f7d5"
+auth_type=sha256
+useconfigure=true
```
Comment (timschumi) [inline Ports/pacman/package.sh:7]: Same here.

```suggestion
auth_type='sha256'
useconfigure='true'
```

File: Ports/pacman/package.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=pacman
+version=git
+workdir=pacman-master
+files="https://github.com/ebuc99/pacman/archive/refs/heads/master.tar.gz pacman-master.tar.gz 3016af69e168e0dd87a079bbfb35d995cf2e7bc1b710716c81ac99323bb4f7d5"
```
Comment (timschumi) [inline Ports/pacman/package.sh:5]: Please lock this to a specific revision, even if it is seems unlikely that the project will get updated.

File: Ports/pacman/package.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+auth_type=sha256
+useconfigure=true
+depends=("SDL2" "SDL2_image" "SDL2_ttf" "SDL2_mixer")
+configopts=(
+    "--with-sdl-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+)
+launcher_name="Pacman"
+launcher_category="Games"
+launcher_command=/usr/local/bin/pacman
+icon_file="data/gfx/pacman_desktop.png"
```
Comment (timschumi) [inline Ports/pacman/package.sh:15]:
```suggestion
launcher_name='Pacman'
launcher_category='Games'
launcher_command='/usr/local/bin/pacman'
icon_file='data/gfx/pacman_desktop.png'
```

File: Ports/pacman/package.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=pacman
+version=git
+workdir=pacman-master
+files="https://github.com/ebuc99/pacman/archive/refs/heads/master.tar.gz pacman-master.tar.gz 3016af69e168e0dd87a079bbfb35d995cf2e7bc1b710716c81ac99323bb4f7d5"
+auth_type=sha256
+useconfigure=true
+depends=("SDL2" "SDL2_image" "SDL2_ttf" "SDL2_mixer")
```
Comment (timschumi) [inline Ports/pacman/package.sh:8]: Should be one line each.

```suggestion
depends=(
    "SDL2"
    "SDL2_image"
    "SDL2_mixer"
    "SDL2_ttf"
)
```

File: Ports/pacman/package.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+configopts=(
+    "--with-sdl-prefix=${SERENITY_INSTALL_ROOT}/usr/local"
+)
+launcher_name="Pacman"
+launcher_category="Games"
+launcher_command=/usr/local/bin/pacman
+icon_file="data/gfx/pacman_desktop.png"
+
+pre_patch() {
+    run ./autogen.sh
```
Comment (timschumi) [inline Ports/pacman/package.sh:18]: This seems like it replaces the `config.sub` in the repository (as the repository itself only features a `config.sub` from 2014, which lacks support for `serenity`). This ultimately makes port buildability depend on the version of autotools that is installed on the host system.

Please enable `use_fresh_config_sub` and point `config_sub_paths` to the correct file to ensure that we always have a sufficiently new `config.sub` file.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/pacman/package.sh (hunk 3 above)
Comment (gunnarbeutner) [reply Ports/pacman/package.sh]: Aye, good point. Should be fixed now. :)

**Review by GMTA — APPROVED** [review]

File: Ports/pacman/package.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='pacman'
+version='b6241a373cc09f021c2ab29714eca5f9e33463f8'
+files="https://codeload.github.com/ebuc99/pacman/zip/${version} pacman.zip d688f75d33c7bf4f217bfcaf0d5ee507fd73bb233d77303927d15b54988f0231"
+auth_type='sha256'
+useconfigure=true
```
Comment (timschumi) [inline Ports/pacman/package.sh:6]:
```suggestion
useconfigure='true'
```

**Review by timschumi — CHANGES_REQUESTED** [review]
One last small nitpick. Other than that, I'm just waiting for one of my checkouts to be ready so I can actually test the port. :^)

File: Ports/pacman/package.sh (hunk 7 above)
Comment (gunnarbeutner) [reply Ports/pacman/package.sh]: Oops, fixed. :)

**Pull request merged**